front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3TaggedEntityDAO.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.model;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.front50.model.tag.TaggedEntity;
+import com.netflix.spinnaker.front50.model.tag.TaggedEntityDAO;
+import rx.*;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class S3TaggedEntityDAO extends S3Support<TaggedEntity> implements TaggedEntityDAO {
+  private final AmazonS3 amazonS3;
+  private final String bucket;
+
+  public S3TaggedEntityDAO(ObjectMapper objectMapper,
+                           AmazonS3 amazonS3,
+                           Scheduler scheduler,
+                           int refreshIntervalMs,
+                           String bucket,
+                           String rootFolder) {
+    super(objectMapper, amazonS3, scheduler, refreshIntervalMs, bucket, (rootFolder + "/tags/").replaceAll("//", "/"));
+
+    this.amazonS3 = amazonS3;
+    this.bucket = bucket;
+  }
+
+  @Override
+  public Set<TaggedEntity> all(String prefix) {
+    ObjectListing bucketListing = amazonS3.listObjects(
+      new ListObjectsRequest(bucket, (rootFolder + prefix).toLowerCase(), null, null, 100)
+    );
+    List<S3ObjectSummary> summaries = bucketListing.getObjectSummaries();
+
+    // TODO-AJ this is naive and inefficient
```

Comment:
You should be able to indicate that there were more than 100.
Even if that means listing 101 and seeing if there is a 101st and only returning the first 100.